### PR TITLE
Feat: SM/StrategyBase Unit Tests + Formatting

### PR DESCRIPTION
**Pull request by ypatil12**
Also removes AVSD/EPM immutable from SM

### CODE REVIEW

File: src/contracts/libraries/SlashingLib.sol [hunk 1]
```diff
@@ -129,11 +133,9 @@ library SlashingLib {
         uint256 newShares = currentShares + addedShares;
 
         // Step 3: Calculate newStakerDepositScalingFactor
-        uint256 newStakerDepositScalingFactor = 
-            newShares
-            .divWad(existingDepositShares + addedShares)
-            .divWad(totalMagnitude)
-            .divWad(uint256(ssf.getBeaconChainScalingFactor()));
+        uint256 newStakerDepositScalingFactor = newShares.divWad(existingDepositShares + addedShares).divWad(
```
Comment (0xClandestine) [inline src/contracts/libraries/SlashingLib.sol:136]: Suggest formatting by hand for readability and use:

```
/// forgefmt: disable-next-item
```

File: src/contracts/mixins/SignatureUtils.sol [hunk 2]
```diff
@@ -71,11 +71,7 @@ abstract contract SignatureUtils is ISignatureUtils {
     }
 
     /// @dev Helper for checking if a signature is valid, reverts if not valid.
-    function _checkIsValidSignatureNow(
-        address signer,
-        bytes32 signableDigest,
-        bytes memory signature
-    ) internal view {
+    function _checkIsValidSignatureNow(address signer, bytes32 signableDigest, bytes memory signature) internal view {
         if (!signer.isValidSignatureNow(signableDigest, signature)) {
```
Comment (0xClandestine) [inline src/contracts/mixins/SignatureUtils.sol:75]: I forgot to use `require` here.

**Review by 0xClandestine — COMMENTED** [review]
Nice, were these not used?

**Comment by ypatil12** [comment]
> Nice, were the immutables not used?
 
 Yup EPM was holdover from M1. AVSD was holdover from when SM called into AVSD for slashing POC

**Comment by stevennevins** [comment]
was the OZ dependency change intentional?

**Comment by ypatil12** [comment]
> was the OZ dependency change intentional?

Yes, we're only on 4.9 now

**Pull request merged**